bridge/ffi/src/c_api.rs

```diff
@@ -0,0 +1,34 @@
+use crate::{bindings::*, utils::*};
+use dsnp_graph_core::api::api::GraphState as GraphStateRust;
+use libc::c_void;
+
+#[no_mangle]
+pub extern "C" fn print_hello_graph() {
+	println!("Hello, Graph!");
+}
+
+// Define a C-compatible representation of GraphState
+#[repr(C)]
+pub struct GraphState {
```

Comment:
I wasn't sure we would need the GraphState defined. I know that we would need to keep the state somewhere and we might need to create a singleton one.

Reply:
We would need c types for GraphState since it is comprised of non primitive types like Vec HashMap, however since user of the api wont directly interact with state but via exposed function, we would just keep a reference to state in C type, encapsulating it

Reply:
But think a singleton would solve it, so good recommendation